app/src/main/java/com/eis/geoCalendar/app/GenericTimedEventManager.java

```diff
@@ -0,0 +1,81 @@
+package com.eis.geoCalendar.app;
+
+import com.eis.geoCalendar.events.EventDatabase;
+import com.eis.geoCalendar.timedEvents.DateTime;
+import com.eis.geoCalendar.timedEvents.TimedEvent;
+import com.eis.geoCalendar.timedEvents.TimedEventManager;
+
+import java.util.ArrayList;
+
+/**
+ * This class manages all scheduled events.
+ *
+ * @param <E> The generic timed event.
+ * @author Francesco Bau' helped by Luca Crema.
+ * @version 1.0
+ * @see com.eis.geoCalendar.app.GenericEventManager
+ * @see com.eis.geoCalendar.timedEvents.TimedEvent
+ * @see com.eis.geoCalendar.timedEvents.TimedEventManager
+ * @since 23/12/2019
+ */
+public class GenericTimedEventManager<E extends TimedEvent> extends GenericEventManager<E> implements TimedEventManager<E> {
+
+    /**
+     * Main constructor, needed to acquire the database of the events.
+     *
+     * @param database The database of the events.
+     */
+    public GenericTimedEventManager(EventDatabase<E> database) {
+        super(database);
+    }
+
+    /**
+     * Searches for all events that have a time earlier than the parameter time
+     *
+     * @param time The time upper bound.
+     * @return An {@link ArrayList} of events with time smaller than {@code time} if there is any, an empty array otherwise.
+     */
+    @Override
+    public ArrayList<E> getEventsBeforeTime(DateTime time) {
+        ArrayList<E> eventsList = new ArrayList<>();
+        for (E event : database.getSavedEvents()) {
+            if (event.getTime().compareTo(time) < 0)
+                eventsList.add(event);
+        }
+        return eventsList;
+    }
+
+    /**
+     * Searches for all events that have a time earlier than the parameter time
+     *
+     * @param time The time lower bound.
+     * @return An {@link ArrayList} of events with time greater than {@code time} if there is any, an empty array otherwise.
+     */
+    @Override
+    public ArrayList<E> getEventsAfterTime(DateTime time) {
```

Comment:
Here you might do the same thing as suggested in the `getEventsBeforeTime` function, basically calling `getEventsBetweenTime` so you don't have duplicated code here

Reply:
As said before, you are totally right, that's a very nice idea indeed...
But here's one issue: is there a limit for the future?

Reply:
As said before, I guess new DateTime(Long.MAX_VALUE) would do the job, right?

Reply:
I guess it would, yeah

Reply:
Ok thank you for the suggestion! 👍

Reply:
Is it better now?